staticDHCPd/staticdhcpdlib/databases/_caching.py

```diff
@@ -163,15 +163,67 @@ def _cacheMAC(self, mac, definition, chained):
          definition.domain_name, definition.domain_name_servers, definition.ntp_servers,
          definition.lease_time
         )
-        
+
+class MemcachedCache(_DatabaseCache):
+    """
+    A memory database cache using memcache.
+    """
+    _mac_cache = None #: A dictionary of cached MACs
+    _subnet_cache = None #: A dictionary of cached subnet/serial data
+
+    def __init__(self, name, memcached_server, memcached_age_time, chained_cache=None):
+        """
+        Initialises the cache.
+
+        :param basestring name: The name of the cache.
+        :param basestring memcached_server: Address and port to connect to the memcached server.
+        :param basestring memcached_age_time: number of seconds to store items in memcache.
+        :param :class:`_DatabaseCache <_DatabaseCache>` chained_cache: The next
+            node in the chain; None if this is the end.
+        """
+        _DatabaseCache.__init__(self, name, chained_cache=chained_cache)
+        import memcache
+
+        self.mc_client = memcache.Client([memcached_server])
+        self.memcached_age_time = memcached_age_time
+        _logger.debug("Memcached database-cache initialised")
+
+    def _reinitialise(self):
+        self.mc_client.flush_all()
+
+    def _lookupMAC(self, mac):
+        data = self.mc_client.get(str(mac))
+        if data:
+            (ip, hostname, extra, subnet_id) = data
+            subnet_str = "%s-%s" % subnet_id
+            details = self.mc_client.get(subnet_str)
+            return Definition(
+             ip=ip, lease_time=details[6], subnet=subnet_id[0], serial=subnet_id[1],
+             hostname=hostname,
+             gateways=details[0], subnet_mask=details[1], broadcast_address=details[2],
+             domain_name=details[3], domain_name_servers=details[4], ntp_servers=details[5],
+             extra=extra
+            )
+        return None
+
+    def _cacheMAC(self, mac, definition, chained):
+        subnet_id = (definition.subnet, definition.serial)
+        subnet_str = "%s-%s" % subnet_id
```

Comment:
Is there any validity checking on the subnet for a definition? i.e. are are guaranteed that it can't be a string with spaces before it gets here? because that would make memcache blow up on key store. 
I think serial has to be an int, so we are good there.